### PR TITLE
Adding effective score

### PR DESCRIPTION
**Pull request by amitcha06**

### CODE REVIEW

File: lib/models/match_model.dart [hunk 1]
```diff
@@ -107,3 +109,224 @@ class Match implements HasuraVars {
...
+    level: GridLevel.low,
+  ): 2,
+  EffectiveScore(
+    mode: MatchMode.tele,
+    piece: Gamepiece.cube,
+    level: GridLevel.low,
+  ): 2
+};
+
+double getPoints(final Map<EffectiveScore, double> countedValues) {
```
Comment (amitcha06) [inline lib/models/match_model.dart:210]: Im open for suggestions regarding code dupe in this method

File: lib/models/match_model.dart [hunk 2]
```diff
@@ -107,3 +107,109 @@ class Match implements HasuraVars {
...
+  double points = 0;
+  for (final EffectiveScore effectiveScore in score.keys) {
+    //all of score's EffectiveScore values are aasigned a value when initialized, therefore the '!'.
+    points += countedValues[effectiveScore]! * score[effectiveScore]!;
+  }
+  return points;
+}
```
Comment (amitcha06) [inline lib/models/match_model.dart:215]: nevermind :) 
(if you have better ideas im still open for suggestions)

**Comment by amitcha06** [comment]
@giladwo Itach suggested this code instead of the menual decleration of 'score'. I remember you saying something about 'score' should be menually created so i wanted to run it by you. if you dont like this and prefer the menual decleration, no problem at all

**Review by itamarsch — DISMISSED** [review]

**Review by itamarsch — DISMISSED** [review]

**Review by itamarsch — APPROVED** [review]

**Review by ItayNaumann — APPROVED** [review]
Looks good!

**Pull request merged**